include/irr/asset/IAssetWriter.h

```diff
@@ -59,7 +61,7 @@ enum E_WRITER_FLAGS : uint32_t
 	@see IAssetLoader
 	@see E_WRITER_FLAGS
 */
-class IAssetWriter : public virtual IImageLoader
+class IAssetWriter : public virtual core::IReferenceCounted, public virtual IImageLoader
```

Comment:
its an asset loader!!!!

make a `IImageWriter` that inherits virtually from IReferenceCounted